commands/update-version-in-file/update_version_in_file.go

```diff
@@ -0,0 +1,86 @@
+package updateversioninfile
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/kurtosis-tech/kudet/helpers"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/spf13/cobra"
+	"os"
+	"regexp"
+	"strings"
+)
+
+const (
+	updateVersionInFileCmdStr     = "update-version-in-file <to update filepath> <pattern format string> <new version>"
+	versionRegexStr               = "[0-9A-Za-z_./-]+"
+	formatStrReplacementSubstr    = "%s"
+	expectedNumSearchPatternLines = 1
+)
+
+var versionRegex = regexp.MustCompile(versionRegexStr)
+
+var UpdateVersionInFileCmd = &cobra.Command{
+	Use:   updateVersionInFileCmdStr,
+	Short: "Updates semantic version of a Kurtosis Repo in file",
+	Args:  cobra.ExactArgs(3),
+	RunE:  run,
+}
+
+func run(cmd *cobra.Command, args []string) error {
+	toUpdateFilepath, patternFormatStr, newVersion := args[0], args[1], args[2]
+	fileToUpdateInfo, err := os.Stat(toUpdateFilepath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return stacktrace.Propagate(err, "No file exists at '%s'", toUpdateFilepath)
+		}
+		return stacktrace.Propagate(err, "An error occurred attempting to retrieve file info for file at '%s'", toUpdateFilepath)
+	}
+	fileToUpdateMode := fileToUpdateInfo.Mode()
+	fileToUpdate, err := os.ReadFile(toUpdateFilepath)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred attempting to read file at '%s'", toUpdateFilepath)
+	}
+
+	if !strings.Contains(patternFormatStr, formatStrReplacementSubstr) {
+		return stacktrace.NewError("The replacement substring '%s' was not found in the passed '%s' as required.", formatStrReplacementSubstr, patternFormatStr)
+	}
+
+	if !versionRegex.Match([]byte(newVersion)) {
+		return stacktrace.NewError("The version regex pattern, '%s', was not found in the provided version, '%s'", versionRegexStr, newVersion)
```

Comment:
More accurately, the provided version doesn't match the version regex (remember that regexes are state machines so you wouldn't really say a state machine is "in" a string).

Also same thing with validation being done early, before you start doing logic